src/app/modules/user-management/components/unapproved-users-table/unapproved-users-table.component.ts

```diff
@@ -123,6 +124,9 @@ export class UnapprovedUsersTableComponent
       dialogContentPayload,
     }
 
-    this.dialogService.openDialog(dialogConfig)
+    const currentDialog: MatDialogRef<any> = this.dialogService.openDialog(dialogConfig)
+    currentDialog.afterClosed().subscribe(() => {
```

Comment:
encapsulate this in a subscription and destroy them: this.subscriptions.unsubscribe()

Reply:
Done.